WordPress/UITestsFoundation/TestObserver.swift

```diff
@@ -0,0 +1,63 @@
+import XCTest
+
+class TestObserver: NSObject, XCTestObservation {
+    override init() {
+            super.init()
+            XCTestObservationCenter.shared.addTestObserver(self)
+        }
+
+    func testBundleWillStart(_ testBundle: Bundle) {
+        // Code added here will run only once before all the tests
+        executeWithRetries(disableAutoFillPasswords)
+    }
+
+    func executeWithRetries(_ operation: () -> Bool) {
+        var retryCount = 3
+
+        while !operation() && retryCount > 0 {
+            retryCount -= 1
+        }
+    }
+
+    func disableAutoFillPasswords() -> Bool {
+        let settings = XCUIApplication(bundleIdentifier: "com.apple.Preferences")
+        settings.terminate()
```

Comment:
Do we need this `.terminate()` here? If this will be run once at the start of the test run, is this necessary? I'm thinking there's nothing to terminate at that point but I could be missing something.

Reply:
This is needed in a retry case if something goes wrong in the first execution and also to make our lives easier when running multiple times locally. =]

I added a comment in [`798de02`](https://github.com/wordpress-mobile/WordPress-iOS/pull/21148/commits/798de02aed721e6a34602d00ef8363ef49aee88c).